test/unit/domain/bitcoin/lightning/invoice-expiration.spec.ts

```diff
@@ -1,15 +1,63 @@
+import { SECS_PER_10_MINS, SECS_PER_DAY } from "@config"
+
+import { toSeconds } from "@domain/primitives"
+import { WalletCurrency } from "@domain/shared"
 import { invoiceExpirationForCurrency } from "@domain/bitcoin/lightning"
 
-describe("invoice-expiration", () => {
-  it("BTC invoice expires in 1 days", () => {
-    const creationDate = new Date("2000-01-01T00:00:00.000Z")
-    const expiresAt = invoiceExpirationForCurrency("BTC", creationDate)
-    expect(expiresAt).toEqual(new Date("2000-01-02T00:00:00.000Z"))
+describe("invoiceExpirationForCurrency", () => {
+  const BTC = WalletCurrency.Btc
+  const USD = WalletCurrency.Usd
+  const now = new Date("2000-01-01T00:00:00Z")
+
+  it("should return expiration for BTC currency with default delay", () => {
+    const expectedExpiration = new Date("2000-01-02T00:00:00.000Z")
+    let expiresAt = invoiceExpirationForCurrency(BTC, now)
+    expect(expiresAt).toEqual(expectedExpiration)
+
+    let delay = toSeconds(59)
+    expiresAt = invoiceExpirationForCurrency(BTC, now, delay)
+    expect(expiresAt).toEqual(expectedExpiration)
+
+    delay = toSeconds(0)
+    expiresAt = invoiceExpirationForCurrency(BTC, now, delay)
+    expect(expiresAt).toEqual(expectedExpiration)
+
+    delay = toSeconds(2 * SECS_PER_DAY)
+    expiresAt = invoiceExpirationForCurrency(BTC, now, delay)
+    expect(expiresAt).toEqual(expectedExpiration)
+  })
+
+  it("should return expiration for USD currency with default delay", () => {
```

Comment:
ok. but why do we allow the override of the default value then here https://github.com/GaloyMoney/galoy/pull/2675/files#r1206365122?

Reply:
not sure to understand, the default value that we have right now became the max value.. so for example for USD invoices could be from 1 to 5 minutes (which is even better for us) and for btc invoices from 1 min to 24 hours.

Reply:
ok then that's good